lib/ex_sdp/media.ex

```diff
@@ -44,6 +44,42 @@ defmodule ExSDP.Media do
           attributes: [binary()]
         }
 
+  @spec new(
+          type :: type(),
+          ports :: [:inet.port_number()],
+          protocol :: binary(),
+          fmt :: binary() | [0..127]
+        ) :: t()
+  def new(type, ports, protocol, fmt) do
+    %__MODULE__{
+      type: type,
+      ports: ports,
+      protocol: protocol,
+      fmt: fmt
+    }
+  end
+
+  @spec set_connection_data(media :: t(), connection_data :: ConnectionData.t()) :: t()
+  def set_connection_data(media, connection_data) do
+    Bunch.Struct.put_in(media, :connection_data, connection_data)
+  end
+
+  @spec add_attribute(media :: t(), attribute :: Attribute.t()) :: t()
+  def add_attribute(media, attribute) do
+    attributes = media.attributes ++ [attribute]
+    Bunch.Struct.put_in(media, :attributes, attributes)
+  end
+
+  @spec get_attribute(media :: t(), key :: String.t()) :: Attribute.t()
+  def get_attribute(media, key) do
+    media.attributes
+    |> Enum.find(fn
+      {k, _v} -> k == key
+      # for flag attributes
+      k -> k == key
+    end)
```

Comment:
- this won't find the struct attributes
- `get_attributes` finding all attributes with a given key would be useful too